pkg/docker/util.go

```diff
@@ -155,3 +159,85 @@ func parseRepositoryTag(repos string) (string, string, string) {
 	}
 	return repos, "", ""
 }
+
+// PullImage pulls the Docker image specifies by name taking the pull policy
+// into the account.
+// TODO: The 'force' option will be removed
+func PullImage(name string, d Docker, policy api.PullPolicy, force bool) (*PullResult, error) {
+	// TODO: Remove this after we deprecate --force-pull
+	if force {
+		policy = api.PullAlways
+	}
+
+	// When pull policy not specified, default to DefaultPullPolicy
+	if len(policy) == 0 {
+		policy = api.DefaultBuilderPullPolicy
```

Comment:
It is not needed as we run PullImage() before so the image should always be
available or this should error out.

On Fri, Nov 6, 2015 at 2:14 PM, Maciej Szulik <notifications@github.com>
wrote:

> In pkg/docker/util.go
> <https://github.com/openshift/source-to-image/pull/324#discussion_r44134559>
> :
>
> > @@ -155,3 +159,85 @@ func parseRepositoryTag(repos string) (string, string, string) {
> >  	}
> >  	return repos, "", ""
> >  }
> > +
> > +// PullImage pulls the Docker image specifies by name taking the pull policy
> > +// into the account.
> > +// TODO: The 'force' option will be removed
> > +func PullImage(name string, d Docker, policy api.PullPolicy, force bool) (*PullResult, error) {
> > +	// TODO: Remove this after we deprecate --force-pull
> > +	if force {
> > +		policy = api.PullAlways
> > +	}
> > +
> > +	// When pull policy not specified, default to DefaultPullPolicy
> > +	if len(policy) == 0 {
> > +		policy = api.DefaultBuilderPullPolicy
>
> I'd again suggest creating defaults, but I remember you've it already in
> the followup PR [image: :wink:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/source-to-image/pull/324/files#r44134559>.
>



-- 
-----------------------------------------------
Michal Fojtik <mi@mifo.sk <mfojtik@mifo.sk>>
Red Hat OpenShift, Engineering